Src/Sarif.PatternMatcher/ValidatorsCache.cs

```diff
@@ -210,7 +289,13 @@ public static void DisableValidationCaching(MethodInfo shouldCacheMethodInfo, bo
                     ValidationState = ValidationState.ValidatorNotFound,
                 };
 
-                if (context.TargetUri.IsAbsoluteUri)
+                validationResult.Fingerprint = SearchSkimmer.CreateFingerprintFromMatch(groups);
```

Comment:
>CreateFingerprintFromMatch [](http://example.com/codeflow?start=61&length=26)

we automatically generate a fingerprint now for JSON only rules with no validators. If they use group names that match fingerprint property names, these will flow directly to a fingerprint! Cool feature that enables the JSON only scenario....